packages/next/next-server/server/api-utils.ts

```diff
@@ -55,9 +55,21 @@ export async function apiResolver(
     apiRes.json = data => sendJson(apiRes, data)
 
     const resolver = interopDefault(resolverModule)
+    let wasPiped = false
+
+    if (process.env.NODE_ENV !== 'production') {
+      // listen for pipe event and don't show resolve warning
+      const handlePipe = () => {
+        wasPiped = true
+        res.removeListener('pipe', handlePipe)
+      }
+      res.addListener('pipe', handlePipe)
```

Comment:
How about doing this?
```suggestion
      res.once('pipe', () => (wasPiped = true))
```

Reply:
Looks good to me 👍